telosnetwork/telos-wallet
# #605 | Revoke / Edit Allowances UI

# Fixes #605

## Description
This PR adds the UI (non-functional) for editing and revoking allowances

Note: we cannot currently test single ERC721 tokens as the indexer returns slightly malformed data for them; we will need to test those in a later PR, which I will include in the testing notes

## Test scenarios
- run the code locally on mainnet: `git fetch --all && git checkout 605-revoke-allowances-ui && echo 'NETWORK="mainnet"' > .env && yarn dev`
- go to http://localhost:8081
- log in with the team account (`0x...eE37`)
- go to the approvals page
    - you should see checkboxes in the first column of the table
    - you should see edit icons next to the allowance values in the table
- try various combinations of toggling individual rows and using the master toggle at the top
    - the checkboxes should behave as your intuition would expect
- go to another page of allowance results
    - the checkboxes, including the master one at the top, should reset
- click the edit allowance icon on an erc20 token row with an allowance which is neither unlimited nor 0
    - the allowance modal should show the right token information
    - the "Custom" radio should be selected
    - the allowance amount should be pre-populated in the currency input
    - the Confirm button should be disabled
- open dev tools
- change the allowance amount either by clicking a different radio button or typing a custom value
    - the Confirm button should become enabled
- click the confirm button
    - you should see the new allowance amount printed in the console
- the Cancel button should close the modal
- click the edit allowance icon on an erc20 token row which has an Unlimited allowance
    - the Unlimited radio should be selected when the edit modal opens up
- repeat the steps from the Custom allowance notes above
    - the modal should work the same
- click the edit allowance button on a row which has an NFT
    - the text in the modal should change slightly to say "Collection"
    - the "allowed" radio button should be selected
- click "not allowed"
    - `false` should print in the console

## Checklist:
